src/Neuron/Plugin/Plugins/Links.hs

```diff
@@ -327,15 +328,20 @@ wikiLinkSpec =
             -- Cf link: [[...]]
             cmAutoLink OrdinaryConnection <$> P.try (wikiLinkP 2)
           ]
-    wikiLinkP :: Monad m => Int -> P.ParsecT [CM.Tok] s m Text
+    wikiLinkP :: Monad m => Int -> P.ParsecT [CM.Tok] s m (Text, Maybe Text)
     wikiLinkP n = do
       void $ M.count n $ symbol '['
-      s <- fmap CM.untokenize $ some $ noneOfToks [CM.Symbol ']', CM.Symbol '[', CM.LineEnd]
+      toks <- some $ noneOfToks [CM.Symbol ']', CM.Symbol '[', CM.LineEnd]
       void $ M.count n $ symbol ']'
-      pure s
-    cmAutoLink :: CM.IsInline a => Connection -> Text -> a
-    cmAutoLink conn url =
-      CM.link url title $ CM.str url
+      let parts = CM.untokenize <$> wordsBy (\x -> CM.tokType x == CM.Symbol '|') toks
+      case parts of
+        [url] -> pure (url, Nothing)
+        [url, name] -> pure (url, Just name)
+        [] -> fail "Empty wiki-link encountered"
+        _ -> fail ("Multiple pipe ('|') characters encountered; here are the parts: " ++ T.unpack (T.intercalate ", " parts))
```

Comment:
I'd think sometimes the user might want to use "foo | bar" as the custom text, so in theory we can allow everything to the right of the first `|` as the custom text. But let's merge the PR in its current state since this is an improvement. 

In emanote, as an aside this is how we do it: https://github.com/srid/ema/blob/d8e14ab63a9b25f44591c38c929b04b957f1546d/src/Ema/Helper/Markdown.hs#L206-L211